ui/v2.5/src/components/Settings/Tasks/DataManagementTasks.tsx

```diff
@@ -232,6 +133,45 @@ export const DataManagementTasks: React.FC<IDataManagementTasks> = ({
     return <ImportDialog onClose={() => setDialogOpen({ import: false })} />;
   }
 
+  function renderCleanDialog() {
+    if (dialogOpen.cleanAlert || dialogOpen.clean) {
+      let msg: string;
+      if (cleanOptions.dryRun) {
+        msg = intl.formatMessage({ id: "actions.tasks.dry_mode_selected" });
+      } else {
+        msg = intl.formatMessage({ id: "actions.tasks.clean_confirm_message" });
+      }
+
+      return (
+        <DirectorySelectionDialog
+          allowPathSelection={dialogOpen.clean}
+          message={msg}
+          header={intl.formatMessage({ id: "actions.clean" })}
+          icon={faTrashAlt}
+          acceptButtonText={intl.formatMessage({ id: "actions.clean" })}
+          acceptButtonVariant="danger"
+          onClose={(p) => {
+            // undefined means cancelled
+            if (p !== undefined) {
+              if (dialogOpen.cleanAlert) {
+                // don't provide paths
+                onClean();
+              } else {
+                onClean(p);
+              }
+            }
+
+            setDialogOpen({
+              clean: false,
+              cleanAlert: false,
+            });
+          }}
+        />
+      );
+    }
+    return;
+  }
```

Comment:
Why couldn't this have been put into the existing `CleanDialog` instead of removing the `CleanDialog` altogether?

Reply:
First I duplicated the `CleanDialog` code over into `LibraryTasks.tsx` as an `AutoTagDialog`.
That was almost identical code, and now a third instance of the `DirectorySelectionDialog` code. So I pulled `CleanDialog` and `AutoTagDialog` out into a new component file.
This new file was almost identical to `DirectorySelectionDialog`, except for the extra styling options, message and option to hide the directory selection.

I thought about adding the styling options to `DirectorySelectionDialog` and replacing the directory code in my new file with that component, but I would have to deconstruct `DirectorySelectionDialog` because it already has a `<ModalComponent>` that I couldn't wrap in my higher level `<ModelComponent>`.

I figured since I was already going to add all the styling options to it, adding message and the option to totally hide it was less complicated and less error prone than trying to refactor `DirectorySelectionDialog` and pull out the directory selection code into a sub component that could be imported into an `OptionalDirectorySelectionDialog` component.

I could do that if you think it would be a better approach?

Reply:
> First I duplicated the CleanDialog code over into LibraryTasks.tsx as an AutoTagDialog.
> That was almost identical code, and now a third instance of the DirectorySelectionDialog code. So I pulled CleanDialog and AutoTagDialog out into a new component file.
> This new file was almost identical to DirectorySelectionDialog, except for the extra styling options, message and option to hide the directory selection.

Please do this instead for now. We can address the duplicate code later if needed.